pkg/sql/sem/tree/name_resolution.go

```diff
@@ -308,19 +308,22 @@ func ResolveExisting(
 		// Two parts: D.T.
 		// Try to use the current database, and be satisfied if it's sufficient to find the object.
 		//
-		// Note: we test this even if curDb == "", because CockroachDB
-		// supports querying virtual schemas even when the current
-		// database is not set. For example, `select * from
-		// pg_catalog.pg_tables` is meant to show all tables across all
-		// databases when there is no current database set.
-
-		if found, objMeta, err := r.LookupObject(ctx, lookupFlags, curDb, scName, u.Object()); found || err != nil {
-			if err == nil {
-				namePrefix.CatalogName = Name(curDb)
-				namePrefix.SchemaName = Name(scName)
+		// Note: CockroachDB supports querying virtual schemas even when the current
+		// database is not set. For example, `select * from pg_catalog.pg_tables` is
+		// meant to show all tables across all databases when there is no current
+		// database set. Therefore, we test this even if curDb == "", as long as the
+		// schema name is for a virtual schema.
+
+		if _, isVirtualSchema := sessiondata.VirtualSchemaNames[scName]; isVirtualSchema && curDb == "" {
```

Comment:
i'm confused why we have the `curDb == ""` check here, we want to check if `curDb != ""` too right?

Reply:
yeah i have a mistake here. we should use this first name resolution logic `if curDb != "" || (curDb == "" && isVirtualSchema)`